### PR TITLE
Added a Github CODEOWNERS file.

### PR DESCRIPTION
**Pull request by moorepants**
This file sets up automatic review requests for specific files and
directories in the SymPy repository. See
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners
for more information about the file and its purpose. I've added myself
to be requested to review on any PRs that affect files in physics/vector
and physics/mechanics.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/sympy/physics/mechanics/ @moorepants
+/sympy/physics/vector/ @moorepants
```
Comment (oscarbenjamin) [inline CODEOWNERS:2]:
```suggestion
/sympy/physics/vector/ @moorepants
/sympy/solvers/ode/ @oscarbenjamin
```

**Comment by oscarbenjamin** [comment]
I've added myself for the ode module.

**Comment by asmeurer** [comment]
Maybe add a comment to the top of the file pointing to the documentation, and explaining that anyone should feel free to add themselves.

**Comment by moorepants** [comment]
I've added a comment in the file with some explanation.

**Pull request merged**